src/index.ts

```diff
@@ -1,4 +1,13 @@
+import log from "npmlog";
+
 import server from "./server";
-import { getPort } from "./config";
+import { getPort, getProxyHostPort } from "./config";
+let hostPort;
+try {
```

Comment:
this should be part of `getProxyHostPort` IMHO.

also, the way this file evolves, it should be something like `./bin/cli.js` instead of `index.js` (IMHO).

We should build something like this: https://github.com/conventional-changelog/standard-version/blob/master/bin/cli.js

Reply:
Why should it be part of getProxyHostPort? The way I think about it is that I try to get the proxy host port, which I think should exist as the user should provide that information. If the information is malformed or missing, that's an error. The try catch is only to format errors in general, we could also add the server call here.

Reply:
The way i think this should work is:
- `server.ts` is the server itself, which gets a config and does _not_ provide any defaults
  - this could leverage independent `proxy-server.ts` and `mock-server.ts` parts
- `index.ts` (or `cli.ts`) is our "default" entrypoint which can provide some defaults and gets its parameters from _somewhere_ (env vars or params). 

having part of the logic in `index.ts` and other parts in a `config.ts`, which is solely used by the `index.ts` entrypoint, feels a bit like ripping apart the "parameter discovery" part - which should be in one place IMHO.

referencing `standard-version` again our `index.ts` would be `cli.js`, our `config.ts` is `command.js` and our `server.ts` is `index.js` in `standard-version` code

Reply:
Thank you for taking the time to plot that out for me.
I think about it differently:

- `config.ts` is the central place where the config is picked up. This allows us to later extend it with different config adapters, e.g. pick up the information from a file or the package.json depending on CLI flags
- `index.js` wires the configuration together with the server.
- `server.js` is a secured realm. There should be no doubt that every configuration being set is valid. By this we free this part of the application from checks if a param is there, we just know

I agree that an object might be better suited with a growing config base, but I see no point in changing it now, two params is still perfectly fine and it is not hard to change it later on